### PR TITLE
sql: change IGNORE COLUMNS MySQL option to EXCLUDE COLUMNS

### PR DESCRIPTION
**Pull request by benesch**
We want to align this option with the forthcoming analogous option for sinks (#27758). In the context of sinks, "EXCLUDE" reads better than "IGNORE", because the columns aren't wholly ignored. Rather, they're allowed to be referenced in the PARTITION BY expression and the HEADERS option--they're just *excluded* from the final value.

EXCLUDE also nicely aligns with the `SELECT ... EXCLUDE` syntax that was recently added to DuckDB and Snowflake.

For backwards compatibility, the `IGNORE COLUMNS` option for the MySQL source continues to be accepted as an alias for `EXCLUDE COLUMNS`.

### Motivation

   * This PR refactors existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: doc/user/sql-grammar/sql-grammar.bnf [hunk 1]
```diff
@@ -232,6 +232,7 @@ create_source_mysql ::=
   'FROM' 'MYSQL' 'CONNECTION' connection_name
   ( ( '(' 'TEXT COLUMNS' ('(' (column_name) ( ( ',' column_name ) )* ')')? )? )
       ( ( ',' 'IGNORE COLUMNS' ('(' (column_name) ( ( ',' column_name ) )* ')')? ')' )? )
+      ( ( ',' 'EXCLUDE COLUMNS' ('(' (column_name) ( ( ',' column_name ) )* ')')? ')' )? )
```
Comment (morsapaes) [inline doc/user/sql-grammar/sql-grammar.bnf:235]:
```suggestion
      ( ( ',' ('IGNORE COLUMNS' | 'EXCLUDE COLUMNS' ) ('(' (column_name) ( ( ',' column_name ) )* ')')? ')' )? )
```

**Review by morsapaes — COMMENTED** [review]
Left a minor suggestion for the railroad diagram, but syntax and documentation look good to me.

**Review by ParkMyCar — APPROVED** [review]
LGTM from an Adapter and SQL council perspective

File: src/sql-parser/src/parser.rs [hunk 2]
```diff
@@ -4446,7 +4449,7 @@ impl<'a> Parser<'a> {
                 TableFromSourceOption {
                     name: match *keyword {
                         TEXT => TableFromSourceOptionName::TextColumns,
-                        IGNORE => TableFromSourceOptionName::IgnoreColumns,
+                        IGNORE => TableFromSourceOptionName::ExcludeColumns,
```
[the remark this answers is not in the document]
Comment (benesch) [reply src/sql-parser/src/parser.rs:4452]: Good catch, thanks! Happily it seems the tests caught this error too, so we have good coverage here.

File: test/mysql-cdc/35-ignore-columns.td [hunk 3]
```diff
@@ -37,14 +37,14 @@ contains: unsupported type
```
[the remark this answers is not in the document]
Comment (benesch) [reply test/mysql-cdc/35-ignore-columns.td:37]: Good eye, done.

**Review by rjobanp — APPROVED** [review]

File: doc/user/sql-grammar/sql-grammar.bnf (hunk 1 above)
Comment (benesch) [reply doc/user/sql-grammar/sql-grammar.bnf:235]: Ah sorry, I missed this suggestion. Going to merge this while CI is green, but I'll circle back on this next Thursday when I remove the `IGNORE COLUMNS` from the syntax diagram entirely.

**Pull request merged**